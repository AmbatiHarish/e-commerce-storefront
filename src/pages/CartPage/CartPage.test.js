import React from 'react';
import { render, screen } from '@testing-library/react';
import CartPage from './CartPage';

jest.mock('../../components/Cart/Cart', () => () => <div data-testid="cart-component">Cart Component</div>);

describe('CartPage Component', () => {
  test('renders CartPage and Cart component', () => {
    render(<CartPage />);

    // Check that CartPage renders successfully
    expect(screen.getByTestId('cart-component')).toBeInTheDocument();
    expect(screen.getByText('Cart Component')).toBeInTheDocument();
  });

  test('matches the snapshot', () => {
    const { asFragment } = render(<CartPage />);
    expect(asFragment()).toMatchSnapshot();
  });
});
