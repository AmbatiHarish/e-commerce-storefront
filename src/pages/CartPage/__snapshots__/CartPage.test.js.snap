// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CartPage Component matches the snapshot 1`] = `
<DocumentFragment>
  <div>
    <div
      data-testid="cart-component"
    >
      Cart Component
    </div>
  </div>
</DocumentFragment>
`;
